openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ParametersGenerator.java

```diff
@@ -364,12 +398,30 @@ private Node handleOptionalQueryParameter(Schema<?> schema, NodeList<AnnotationN
         }
     }
 
+    private Node handleReferencedQueryParameter(Parameter parameter, String refTypeName, Schema<?> refSchema,
+                                                NodeList<AnnotationNode> annotations, IdentifierToken parameterName) {
+        BuiltinSimpleNameReferenceNode rTypeName = createBuiltinSimpleNameReferenceNode(null,
+                createIdentifierToken(refTypeName));
+        if (refSchema.getDefault() != null) {
+            String defaultValue = refSchema.getType().equals(GeneratorConstants.STRING) ?
+                    String.format("\"%s\"", refSchema.getDefault().toString()) : refSchema.getDefault().toString();
+            return createDefaultableParameterNode(annotations, rTypeName, parameterName,
+                    createToken(SyntaxKind.EQUAL_TOKEN),
+                    createSimpleNameReferenceNode(createIdentifierToken(defaultValue)));
+        } else if (parameter.getRequired() && (refSchema.getNullable() == null || (!refSchema.getNullable()))) {
+            return createRequiredParameterNode(annotations, rTypeName, parameterName);
+        } else {
+            OptionalTypeDescriptorNode optionalNode = createOptionalTypeDescriptorNode(rTypeName,
+                    createToken(SyntaxKind.QUESTION_MARK_TOKEN));
+            return createRequiredParameterNode(annotations, optionalNode, parameterName);
+        }
+    }
```

Comment:
Missing new line

Reply:
Updated in https://github.com/ballerina-platform/openapi-tools/pull/1335/commits/227a2678799ce9dbf3cdad98015df7107add3b4d